docarray/array/mixins/evaluation.py

```diff
@@ -74,8 +74,7 @@ def evaluate(
 
             binary_relevance = [1 if hash_fn(m) in desired else 0 for m in d.matches]
 
-            if 'max_rel' not in kwargs:
-                kwargs['max_rel'] = len(gd.matches)
+            kwargs['max_rel'] = len(gd.matches)
```

Comment:
I still think max rel needs to be respected. I think we need the logic to tell if max_rel was in kwargs from the beginning or not

Reply:
if we have `max_rel` in original args it means that the user assumes is the same for all `docs`

Reply:
Otherwise u do not give this power to the user